scalardb-test/src/main/java/kelpie/scalardb/transfer/TransferProcessorWithRetry.java

```diff
@@ -0,0 +1,108 @@
+import java.util.concurrent.atomic.LongAdder;
+import javax.json.Json;
+
+public class TransferProcessorWithRetry extends TimeBasedProcessor {
+
+  private static final int INITIAL_INTERNAL_MILLS = 10;
+  private static final double MULTIPLIER = 2.0;
+  private static final int MAX_RETRIES = 10;
+
+  private final DistributedTransactionManager manager;
+  private final int numAccounts;
+  private final Retry retry;
+  private final LongAdder retryCount = new LongAdder();
+
+  public TransferProcessorWithRetry(Config config) {
+    super(config);
+    this.manager = TransferCommon.getTransactionManager(config);
+    this.numAccounts = (int) config.getUserLong("test_config", "num_accounts");
+
+    RetryConfig retryConfig =
+        RetryConfig.custom()
+            .maxAttempts(MAX_RETRIES)
+            .intervalFunction(
+                IntervalFunction.ofExponentialBackoff(INITIAL_INTERNAL_MILLS, MULTIPLIER))
+            .retryExceptions(CrudConflictException.class, CommitConflictException.class)
+            .build();
+    retry = Retry.of("TransferProcessorWithRetry", retryConfig);
+  }
+
+  @Override
+  public void executeEach() throws Exception {
+    int fromId = ThreadLocalRandom.current().nextInt(numAccounts);
+    int toId = ThreadLocalRandom.current().nextInt(numAccounts);
+    int amount = ThreadLocalRandom.current().nextInt(1000) + 1;
+
+    try {
+      Retry.decorateCheckedRunnable(retry, () -> transfer(fromId, toId, amount)).run();
+    } catch (Throwable t) {
+      if (t instanceof Exception) {
+        throw (Exception) t;
+      }
+
+      throw new Exception(t);
+    }
+  }
+
+  @Override
+  public void close() {
+    manager.close();
+
+    setState(Json.createObjectBuilder().add("retry_count", retryCount.longValue()).build());
+  }
+
+  private void transfer(int fromId, int toId, int amount) throws Exception {
+    int fromType = 0;
+    int toType = 0;
+    if (fromId == toId) {
+      toType = 1; // transfer between the same account
+    }
+
+    DistributedTransaction transaction = manager.begin();
+    try {
+      Get fromGet = TransferCommon.prepareGet(fromId, fromType);
+      Get toGet = TransferCommon.prepareGet(toId, toType);
+
+      Optional<Result> fromResult = transaction.get(fromGet);
+      Optional<Result> toResult = transaction.get(toGet);
+      int fromBalance = TransferCommon.getBalanceFromResult(fromResult.get());
+      int toBalance = TransferCommon.getBalanceFromResult(toResult.get());
+
+      Put fromPut = TransferCommon.preparePut(fromId, fromType, fromBalance - amount);
+      Put toPut = TransferCommon.preparePut(toId, toType, toBalance + amount);
+      transaction.put(fromPut);
+      transaction.put(toPut);
+
+      transaction.commit();
+    } catch (Exception e) {
+      if (e instanceof CrudConflictException || e instanceof CommitConflictException) {
+        logWarn("a transaction conflict occurred. Retrying...", e);
+        retryCount.increment();
```

Comment:
Count the retry count to show it in the report.

Reply:
The library seems to have EventPublisher that we can inject a callback to increment the counter. I'm not sure if it works, but it might be worth trying.

https://github.com/resilience4j/resilience4j/blob/4514366d418fd08bbdd48f0bab92693862715bdb/README.adoc#71-examples

Reply:
Oh, I didn't know that. Let me check. Thank you.

Reply:
Fixed in 65c672add3ee5dc954b3c5fba9066129b81fb784.

@komamitsu Can you please take a look again when you have time! Thanks.